pkg/daemon/ceph/osd/kms/k8s.go

```diff
@@ -54,6 +60,24 @@ func (c *Config) storeSecretInKubernetes(pvcName, key string) error {
 	return nil
 }
 
+// updateSecretInKubernetes updates the dmcrypt key in a Kubernetes Secret
+func (c *Config) updateSecretInKubernetes(pvcName, key string) error {
+	secretName := GenerateOSDEncryptionSecretName(pvcName)
+	secret, err := c.context.Clientset.CoreV1().Secrets(c.ClusterInfo.Namespace).Get(c.ClusterInfo.Context, secretName, metav1.GetOptions{})
```

Comment:
> Instead of a Get and Update, could we call `generateOSDEncryptedKeySecret()` to create the spec and directly call Update?

This call goes from the key rotation job.

It's better to update the already existing secret(with all the ownerref and the sought set already).

Reply:
> Do we ever expect anything to update the secret except rook? If rook completely owns the resources, it's a nice pattern to generate the same spec whether it's a create or update, but this way certainly works too.

This update runs from the cronjob, creating same spec also means setting ownerref to the cluster cr(needs extra rbac to do so). 
Just fetching and updating only the key section of the secret is simpler.